drivers/storage/portworx/portworx.go

```diff
@@ -299,6 +302,9 @@ func (p *portworx) SetDefaultsOnStorageCluster(toUpdate *corev1.StorageCluster)
 	}
 
 	setDefaultAutopilotProviders(toUpdate)
+	if pxutil.IsCSIEnabled(toUpdate) {
```

Comment:
Should we reset the CSI feature gate and use `cluster.Spec.CSI` in this function?
- If we don't reset then users will have to toggle two flags to disable CSI
- Once we have populated `cluster.Spec.CSI.Enabled`, then rest of the code, should be able to just use the above IsCSIEnabled function which checks the new flag instead of the featureGates map.

Reply:
lets do that change in a separate PR, feel free to open a bug to track that.